### ISSUE TITLE
sourceFromExchange broken

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#760]
There is a crash when running sarra configurations with the sourceFromExchange setting on.  Is the "source" setting a message field or a setting?  requires study.

**Comment by iMacadan** [comment]
In sarra config, the work around for bug should be something like:
#source_from_exchange    True
source                  MSC-DMS-DEV

```
2023-09-27 15:12:44,355 [INFO] 36406 sarracenia.moth.amqp getSetup queue declared q_feeder.sarra.cvt_minify.ddsr-cmc-dev01.cmc.ec.gc.ca (as: amqp://feeder@localhost/)
2023-09-27 15:12:44,355 [INFO] 36406 sarracenia.moth.amqp getSetup binding q_feeder.sarra.cvt_minify.ddsr-cmc-dev01.cmc.ec.gc.ca with v02.post.*.MSC-DMS-DEV.WXAPP.ALERT_GEOJSON.# to xpublic (as: amqp://feeder@localhost/)
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/sarracenia/instance.py", line 242, in <module>
    i.start()
  File "/usr/lib/python3/dist-packages/sarracenia/instance.py", line 233, in start
    self.running_instance.run()
  File "/usr/lib/python3/dist-packages/sarracenia/flow/__init__.py", line 452, in run
    self.filter()
  File "/usr/lib/python3/dist-packages/sarracenia/flow/__init__.py", line 978, in filter
    pstrip, flatten)
  File "/usr/lib/python3/dist-packages/sarracenia/flow/__init__.py", line 847, in updateFieldsAccepted
    new_dir = self.o.variableExpansion(maskDir, msg)
  File "/usr/lib/python3/dist-packages/sarracenia/config.py", line 2066, in variableExpansion
    new_dir = new_dir.replace('${SOURCE}', message['source'])
KeyError: 'source'
2023-09-27 15:12:44,393 36341 [CRITICAL] root run_command subprocess.run failed err=Command '['/usr/bin/python3', '/usr/lib/python3/dist-packages/sarracenia/instance.py', '--no', '0', 'foreground', 'sarra/cvt_minify']' returned non
-zero exit status 1.
```

**Comment by petersilva** [comment]
merged directly to v03_wip by mistake, rather than doing a PR... anyways... problem solved...

**Comment by petersilva** [comment]
released with 3.0.44
